### ISSUE TITLE
use intermediate function for error checking

### ISSUE THREAD

**Issue by psychocoderHPC** [ComputationalRadiationPhysics/picongpu#2672]
As shown in https://github.com/ComputationalRadiationPhysics/picongpu/pull/2666#pullrequestreview-143914963 comment it is not good to use scopes `{}` within a macro. This can have side effects if the macro is used in  a `if/else` condition.

# Todo

Create a intermediate function for the error checking code and call this function directly within the macro and path usefull information like `__FILE__` to the helper function.

**Comment by sbastrakov** [comment]
@psychocoderHPC I agree a function is better, need to decide to call a function from a macro or to call a macro to generate function parameters.
However, to be fair a macro can be safely (in regard to if statements) wrapped in a do-while construct.
